gammapy/maps/axes.py

```diff
@@ -0,0 +1,145 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+
+import numpy as np
+from astropy.time import Time
+
+
+class TimeAxis:
+    """Class representing a time axis.
+
+    Provides methods for transforming to/from axis and pixel coordinates.
+    A time axis can represent non-contiguous sequences of time intervals.
+
+    Parameters
+    ----------
+    time_min : `~astropy.time.Time`
+        Array of edge time values.
+    time_max : `~astropy.time.Time`
+        Array of edge time values.
+    interp : str
+        Interpolation method used to transform between axis and pixel
+        coordinates.  Valid options are 'log', 'lin', and 'sqrt'.
+    name : str
+        Axis name
+    """
+    node_type = "edges"
+    def __init__(self, time_min, time_max, name="", interp="lin",):
+        self._name = name
+
+        invalid = [type(_).__name__ for _ in (time_min, time_max) if not isinstance(_, Time)]
+        if len(invalid)>0:
+            raise TypeError(f"TimeAxis edges must be Time objects. Got {invalid}")
+
+        if not len(time_min) == len(time_max):
+            raise ValueError("Time min and time max must have the same length.")
+
+        if not (np.all(time_min == time_min.sort()) and np.all(time_max == time_max.sort()) ):
+            raise ValueError("TimeAxis: edge values must be sorted")
+
+        self._time_min = Time(time_min)
```

Comment:
It could be advantageous to always go to the same internal representation, it depends a bit on which internal methods we need and whether they support `Time` objects. This is e.g. not the case for interpolation...

Reply:
What is the best approach?
We could also define a `TimeMapAxis.time_ref` and use all times as quantities similarly to what is done for `GTI` objects. Then interpolation is easier.

Reply:
I agree, defining a reference time and relying on this internally is maybe the most promising approach...

Reply:
It is implemented now.